### PR TITLE
new structure of the future BigDL tutorial 

### PR DESCRIPTION
**Pull request by megaSpoon**
new structure of the BigDL tutorial wiki @jason-dai @yangw1234 @yiheng

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,61 @@
...
+   - install dependencies and imports
+   - mechanics of weight and gradient update
+     - Forward and backward(topic 5)
+   - Feedforward Neural Network(topic 9)
+   - RNN(topic 11)
+   - Bi-RNN(topic 13)
+   - LSTM(topic 12)
+   - CNN(topic 10 will include "batch normalization" here)
+     - batch normalization
+   - Auto-encoder(topic 14)
```
Comment (jason-dai) [inline README.md:57]: Maybe we can expand this section, similar to Part 1 of http://gluon.mxnet.io

Comment (jason-dai) [reply README.md:57]: And add Keras-like API implementation as well @zhichao-li

File: README.md [hunk 2]
```diff
@@ -18,21 +73,4 @@ Step-by-step Deep Leaning Tutorials on Apache Spark using [BigDL](https://github
 13. [Bi-directional RNN](https://github.com/intel-analytics/BigDL-Tutorials/blob/master/notebooks/neural_networks/birnn.ipynb)
 14. [Auto-encoder](https://github.com/intel-analytics/BigDL-Tutorials/blob/master/notebooks/neural_networks/autoencoder.ipynb)
 
-### Environment
```
Comment (yiheng) [inline README.md:21]: why remove this?

File: README.md [hunk 3]
```diff
@@ -18,21 +73,4 @@ Step-by-step Deep Leaning Tutorials on Apache Spark using [BigDL](https://github
...
-+ Python 2.7
-+ JDK 8
-+ Apache Spark 2.2.0
-+ Jupyter Notebook 4.1
-+ BigDL 0.3.0
-+ [Setup env on Mac OS](https://github.com/intel-analytics/BigDL-Tutorials/blob/master/SetupMac.md) / [Setup env on Linux](https://github.com/intel-analytics/BigDL-Tutorials/blob/master/SetupLinux.md)
 
-### Start Jupyter Server
```
Comment (yiheng) [inline README.md:29]: why remove this?

File: README.md [hunk 4]
```diff
@@ -18,21 +73,4 @@ Step-by-step Deep Leaning Tutorials on Apache Spark using [BigDL](https://github
...
-* Run ```pip install BigDL==0.3.0```
-* Run ``` jupyter notebook --notebook-dir=./ --ip=* --no-browser```
-
-## Run Demo
```
Comment (yiheng) [inline README.md:33]: why remove this?

**Comment by yiheng** [comment]
http://gluon.mxnet.io is a document-like website. I think it's different from the purpose of this project. This project provides a list of notebooks on mnist to demonstrate how to use bigdl and user can try by themselves. I strongly suggest you create another repo. Just like our doc website to hold your tutorial.

**Comment by megaSpoon** [comment]
@yiheng Thanks for the review. Basically,  we want to build a document-like website similar to mxnet tutorial page, which has a navigation bar with progressive instructions, enough details and various applications to fully present the flavor of deep learning with BigDL. That's the final goal so we can combine all the notebooks, emphasize their relativeness and provide users more friendly experience. But the current tutorial might be prone to be more single-tasking(MNIST digit recognition) and less elegant. Thus inadvertently, the tutorial needs a better organization with a document-like wiki support if the BigDL tutorial repository is expanded with more other topics in the future. Based on this wish, @jason-dai suggests me to put a proposal of some viable navigation structure just like what we can see in the navigation of http://gluon.mxnet.io. The proposal is temporarily put in the readme for everyone to review so don't worry about the readme change since it's just a proposal for open suggestion  and ideas.

**Comment by megaSpoon** [comment]
Although I wasn't told that the project was focused on MNIST. I can start a new repo. It's just that  before finalizing the tree structure of the directory, we can probably have some reviews and ideas from peers about how to make the tutorial more scalable and think of what meaningful aspect of deep learning our repo is missing. Such aspects like overfitting, regularization, batch normalization which are basic but we can see in the MXNet tutorial they cherry picked almost every fundamental topic to teach novices from bottom to top hand by hand. It forms the quality of its tutorial which tends to be very user-friendly. We can make a better tutorial of the same quality.

File: README.md [hunk 5]
```diff
@@ -2,6 +2,61 @@
...
+  - prologgue
+  - Organization of the tutorials
+  - advices and prerequisites for learners
+ 
+ - Introduction to Spark basics(topic 1-4)
+   - introduction to What Spark is, current usage and application(provided by useful links from Spark official site)
+   - environment setting and install instructions
+   - RDD
+   - DataFrame
+   - SparkSAL
```
Comment (yangw1234) [inline README.md:15]: SparkSQL?

**Comment by jason-dai** [comment]
I think one option is to create another folder for the new tutorials, which can be implemented by porting the MXNet tutorials (https://github.com/zackchase/mxnet-the-straight-dope) to BigDL. In the first stage, we  can just implement these using the Keras APIs in BigDL.
